### PR TITLE
jhead: 3.04 -> 3.06.0.1

### PR DESCRIPTION
**Pull request by LeSuisse**
###### Motivation for this change

Fixes CVE-2020-6624 and CVE-2020-6625.

https://github.com/Matthias-Wandel/jhead/releases/tag/3.06.0.1

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>vimiv</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>image_optim</li>
    <li>jhead</li>
  </ul>
</details>

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119624` at cfce3601 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>vimiv</li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>image_optim</li>
    <li>jhead</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/graphics/jhead/default.nix:16:3:
```
   |
16 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/tools/graphics/jhead/default.nix [hunk 1]
```diff
@@ -1,31 +1,18 @@
...
-  version = "3.04";
+  version = "3.06.0.1";
 
-  src = fetchurl {
-    url = "http://www.sentex.net/~mwandel/jhead/${pname}-${version}.tar.gz";
-    sha256 = "1j831bqw1qpkbchdriwcy3sgzvbagaj45wlc124fs9bc9z7vp2gg";
+  src = fetchFromGitHub {
+    owner = "Matthias-Wandel";
+    repo = "jhead";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/jhead/default.nix:10]:
```suggestion
    rev = version;
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 119624` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>vimiv</li>
</ul>
</details>
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>image_optim</li>
<li>jhead</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM.

Not sure if we want to retain ``CFLAGS=-O3``.

**Comment by LeSuisse** [comment]
> Not sure if we want to retain CFLAGS=-O3.

I updated the PR to keep the flags.

**Pull request merged**